src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -427,10 +576,53 @@ void EIRPlantLoopHeatPump::doPhysics(EnergyPlusData &state, Real64 currentLoad)
                                        capacityModifierFuncTemp);
         capacityModifierFuncTemp = 0.0;
     }
+}
```

Comment:
New member function for checking the capacity modifier curve value, resets the value to 0.0 if negative and throws error message.